packages/fxa-settings/src/models/contexts/AppContext.ts

```diff
@@ -24,13 +26,80 @@ export interface AppContextValue {
   config?: Config;
   account?: Account;
   session?: Session; // used exclusively for test mocking
+  uniqueUserId?: string; // used for experiments
 }
 
 export interface SettingsContextValue {
   alertBarInfo?: AlertBarInfo;
   navigatorLanguages?: readonly string[];
 }
 
+/**
+ * Fetches or generates a new client ID that is stable for that browser client/cookie jar.
+ *
+ * N.B: Implemenation is taken from `fxa-content-server/.../models/unique-user-id.js` with
+ * inlined code that was written using Backbone utilities which could not immediately be transferred over.
+ * @returns a new or existing UUIDv4 for this user.
+ */
+function getUniqueUserId(): string {
```

Comment:
Placed this here for now, but I can move it to some other place.

Reply:
Yeah I think this function should go in our `fxa-settings/src/lib/cache.ts` file especially if we don't need the resume token stuff (see comment below), check that out and see how we are getting values from local storage as well since we should avoid directly getting/setting on it. (We have a ticket in the React Cleanup epic to make our local storage references consistent FWIW, I'm sure we are doing this in at least a few other places.)

Reply:
Ah thanks! I somehow missed that we can use the Storage abstraction from content-server in settings too.